libobs/graphics/graphics-ffmpeg.c

```diff
@@ -270,12 +273,21 @@ uint8_t *gs_create_texture_file_data(const char *file,
 	uint8_t *data = NULL;
 
 	if (ffmpeg_image_init(&image, file)) {
-		data = bmalloc(image.cx * image.cy * 4);
+		*cx_out = (uint32_t)image.cx;
+		*cy_out = (uint32_t)image.cy;
+		const int image_resolution_limit = 10000*10000;
+		if ((image.cx * image.cy > image_resolution_limit)) {
+			blog(LOG_WARNING, "Image resolution over 100MP limit and get downscaled");
+			float devider = (image.cx * image.cy) / image_resolution_limit;
+			*cx_out = (uint32_t)(image.cx/devider);
+			*cy_out = (uint32_t)(image.cy/devider);
+		}
+
+		data = bmalloc((*cx_out) * (*cy_out) * 4);
```

Comment:
Can bmalloc fail?

Reply:
obs implementation of bmalloc throws an exception if it failed to allocate memory.